.github/workflows/ci.yml

```diff
@@ -156,16 +155,32 @@ jobs:
           key: nginx-kubernetes-gateway-${{ github.run_id }}-${{ github.run_number }}
       - name: Docker Buildx
         uses: docker/setup-buildx-action@v1
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        if: ${{ startsWith(github.ref, 'refs/tags') }}
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=ghcr.io/nginxinc/nginx-kubernetes-gateway
+          tags: |
```

Comment:
you might want to add more tags since the workflow runs for every PR and commit to `main` 

I think this would be enough for now
```
type=edge
type=ref,event=pr
```

Reply:
>the workflow won't run unless the tag is a release tag like vx.y.z so that won't run on commits to main

the workflow will also run for every commit to `main` and `release` branches and every PR against main, it just won't push the image to the registry.

By adding those types I mentioned the meta step will just dynamically create a tag based on the event, for example for this PR it would be `pr-146`, `edge` when it's a commit to main.
You would also need something similar to this for the release branch
```
type=ref,event=branch,enable=${{ startsWith(github.ref, 'refs/heads/release') }}
```

By adding the `type=sha` you will have a tag with the sha for all of the events unconditionally. This means that when you push the image on a "tag event", it will also tag and push the image with the sha. 
I don't believe this is what we want?

Reply:
Thank you.  You're right that we don't need the image tagged with sha to be pushed to the registry on a release tag workflow and we don't need the other tags on a main or release branch so I added a couple of enable conditions like the one you suggested here.  Thanks!  

> the workflow will also run for every commit to main and release branches and every PR against main, it just won't push the image to the registry.

Right.  I intentionally configured it to do that.  I was just not being very clear in my explanation.  :)

Thanks again for all your help with this.

Reply:
Now I think it will fail because you're not creating a tag on PRs or the release branch. Really using the one I suggested is just the simpler option
```
type=edge 
type=ref,event=pr
type=ref,event=branch,enable=${{ startsWith(github.ref, 'refs/heads/release') }}
```
the action automatically detects the type of event, commit to `main`, PR, commit to the release branch and creates the name accordingly

Reply:
It doesn't fail.  You can see it creating a tag and passing in this workflow from my fork on a release branch.
```
https://github.com/pammecrandall/nginx-kubernetes-gateway/actions/runs/2730156852
```
and here it is passing and pushing an image to the registry on a release tag.

I appreciate the suggestion but the team doesn't want edge tags or pr tags on the images so this might be simpler but it won't do what we want it to do.

Reply:
After our meeting, I made these updates.
```
          images: |
            name=ghcr.io/pammecrandall/nginx-kubernetes-gateway,enable=${{ startsWith(github.ref, 'refs/tags') }}
            name=nginx/nginx-kubernetes-gateway,enable=${{ !startsWith(github.ref, 'refs/tags') }}
          tags: |
            type=semver,pattern={{version}}
            type=sha,enable=${{ !startsWith(github.ref, 'refs/tags') }}
    
         ```
and
  ```
        - name: Run Trivy vulnerability scanner
        uses: aquasecurity/trivy-action@0.2.3
        continue-on-error: true
        with:
          image-ref: ${{ steps.meta.outputs.tags }}
          ```          
so it will use nginx image and sha tag on branches and the ghcr image with the release tag on release tags.
Here are a couple of successful runs with these changes in my fork.

This one is from a release branch.
```https://github.com/pammecrandall/nginx-kubernetes-gateway/actions/runs/2755268177```

and this one is from a release tag.
```https://github.com/pammecrandall/nginx-kubernetes-gateway/actions/runs/2755269152```

Reply:
I replaced the sha tag after the team agreed to use an edge tag in the recent slack conversation.